contrib/endpoints/src/api_manager/service_control/aggregated.cc

```diff
@@ -442,13 +433,75 @@ Status Aggregated::GetStatistics(Statistics* esp_stat) const {
   return Status::OK;
 }
 
+template <class RequestType>
+const std::string& Aggregated::GetApiReqeustUrl(const RequestType& request) {
+  if (typeid(request) == typeid(CheckRequest)) {
+    return url_.check_url();
+  } else if (typeid(request) == typeid(AllocateQuotaRequest)) {
+    return url_.quota_url();
+  } else {
+    return url_.report_url();
+  }
+}
+
+template <class RequestType>
+void Aggregated::SetHttpRequestTimeout(
+    const RequestType& request, std::unique_ptr<HTTPRequest>& http_request) {
```

Comment:
Remove http_request argument and changed function name to GetHttpRequestTimeout.